sdk/lib/opentelemetry/sdk/trace/samplers/consistent_probability_tracestate.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      module Samplers
+        # @api private
+        #
+        # The ConsistentProbabilityTraceState module implements common TraceState
+        # validation and manipulation for the consistent probability-based samplers.
+        module ConsistentProbabilityTraceState
+          DECIMAL = /\A\d+\z/.freeze
+          private_constant(:DECIMAL)
+
+          def validate_tracestate(span_context) # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
```

Comment:
The method naming here is interesting - we're not really returning boolean indicating if it's valid; we're actually parsing the `ot` bits out of the trace state, validating them, allowing you to modify them, and then serializing it all back to the trace state.

Maybe `with_validated_tracestate`? Not sure really.

Reply:
Update: `with_validated_tracestate` also doesn't make a ton of sense, given that we don't always _expect_ it to be called with a block... maybe `sanitize_tracestate!` ? 🤔

Reply:
It's rough because the spec refers to this as validating tracestate. I totally agree that it doesn't make sense, and `sanitize_tracestate` is clearer, but we'd then need to add comments to tie this back to the spec (so someone reading the spec alongside the code can understand what we're implementing).

Reply:
Technically, we're not mutating the tracestate - we're returning a validated (really, sanitized, as you said) copy of the tracestate extracted from the `span_context`. So the `!` doesn't apply here.

[The implementation for Go](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/samplers/probability/consistent/tracestate.go) is interesting in that they haven't built a generic representation of the OpenTelemetry `tracestate` concern, probably because these are the only keys currently defined - they've built something very specific - essentially just the pair of values. I'm trying to decide if we should do something similar, and if that will simplify the code a little (e.g. get rid of the `Tracestate`/`OpenTelemetryConcern` class).

Reply:
I'm refactoring at the moment based on the Go implementation. I *think* it'll be a little more efficient, and we can revert the refactoring later if/when we add more otel-specific concerns in the tracestate. I hope, when we do that, that we have a clear and consistent API spec for working with those concerns.

Reply:
That makes sense to me!